src/streamlink/plugin/plugin.py

```diff
@@ -402,6 +403,11 @@ def stream_weight_only(s):
             worst = sorted_streams[0]
             final_sorted_streams["worst"] = streams[worst]
             final_sorted_streams["best"] = streams[best]
+        else:
+            best = unfiltered_sorted_streams[-1]
```

Comment:
current behavior **without this PR**

```
error: The specified stream(s) 'best' could not be found.
       Available streams: vod, vod_alt, vod_alt2
```

**with this PR**

as `vod` is not a valid sorted stream name for Streamlink,
**sorted_streams (unfiltered_sorted_streams)** is empty here and it will error

```
Traceback (most recent call last):
  File "venv/bin/streamlink", line 11, in <module>
    load_entry_point('streamlink', 'console_scripts', 'streamlink')()
  File "src/streamlink_cli/main.py", line 1007, in main
    handle_url()
  File "src/streamlink_cli/main.py", line 557, in handle_url
    streams = fetch_streams(plugin)
  File "src/streamlink_cli/main.py", line 437, in fetch_streams
    sorting_excludes=args.stream_sorting_excludes)
  File "src/streamlink/plugin/plugin.py", line 407, in streams
    best = unfiltered_sorted_streams[-1]
IndexError: list index out of range
```

it shouldn't error and it should return the same as before or handle it with some unfiltered worst/best

---

this can be used to test it without a valid URL / Plugin
```diff
--- a/src/streamlink/plugin/plugin.py
+++ b/src/streamlink/plugin/plugin.py
@@ -382,9 +382,11 @@ class Plugin(object):
             return (self.stream_weight(s)[0] or
                     (len(streams) == 1 and 1))
 
+        streams = {'vod': '', 'vod_alt': '', 'vod_alt2': ''}
         stream_names = filter(stream_weight_only, streams.keys())
         sorted_streams = sorted(stream_names, **key=stream_weight_only)**
```